packages/@react-aria/loading/src/useLoadOnScroll.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {RefObject, useCallback, useMemo, useRef} from 'react';
+import {useLayoutEffect} from '@react-aria/utils';
+
+export interface LoadOnScrollProps {
+  /** Whether data is currently being loaded. */
+  isLoading?: boolean,
+  /** Handler that is called when more items should be loaded, e.g. while scrolling near the bottom.  */
+  onLoadMore?: () => void,
+  // TODO: decide on default here
+  /**
+   * The amount of offset (in pixels) from the bottom of your scrollable region that should trigger load more.
+   * @default 25
+   */
+  scrollOffset?: number
+}
+
+// TODO: discuss if it would be ok to just attach event to ref...
+interface LoadOnScrollAria {
+  /** Props for the scrollable region. */
+  scrollViewProps: {
+    onScroll: () => void
+  }
```

Comment:
I shifted this over to be scrollViewProps so it could be used in useVirtualizer (though I guess I could have useVirtualizer just return a empty func for its onVisibleRectChange). I would've preferred attaching the scroll event directly to the ref via useEvent though...